api/envoy/config/route/v3/route_components.proto

```diff
@@ -837,6 +837,62 @@ message RouteAction {
   string prefix_rewrite = 5
       [(validate.rules).string = {well_known_regex: HTTP_HEADER_VALUE strict: false}];
 
+  // Indicates that some headers received from upstream cluster need to be rewritten. When upstream
+  // returns a header, the header will be overwritten with the given value before being sent back as
+  // response to the client.
+  //
+  // In the below example there are two response header rewrites mentioned: one for location header
+  // and one for x-random-token header.
+  //
+  // The location header's value will be matched with the given match pattern as a prefix to the header's
+  // value. If a match exists, the matched portion may be swapped with the value given in substitution.
+  // If no match string is given, the value is replaced entirely with the given substitution. The
+  // response status header is also matched before making a substitution. If the response status
+  // returned from the upstream cluster matches with one of the response statuses in the list, the
+  // substitution can proceed after matching the match section. If this list is empty, response status
+  // returned is not checked before making the substitution.
+  //
+  // The x-random-token header's value will be matched using the regex pattern specified in the pattern
+  // section, and the matched part will be replaced by the value provided in the substitution section.
+  //
+  // If a header is specified in response_headers_to_rewrite section but is not returned from the
+  // upstream it it simply ignored.
+  //
+  //   .. code-block:: yaml
+  //
+  //     - match:
+  //         prefix: "/itm/"
+  //       route:
+  //         response_headers_to_rewrite:
+  //           "location":
+  //             header_rewrite_literal:
+  //               match: /itm/not-found/
+  //               substitution: /n/error
+  //               response_status:
```

Comment:
Status is just one header you could potentially also match one. One thing you could consider doing here for matching is to look at what `ResponseMapper` in HCM does, where it uses `AccessLogFilter` as a general purpose matcher for responses prior to applying an action.

Reply:
I looked at response mapper in HCM. It has a repeated field `headers_to_add`. They are trying to add headers, so the default assumption is that every header must be added.

For me the case is different. I am trying to *match* the headers.
```
repeated config.core.v3.HeaderValue upstream_headers_matcher = 3
      [(validate.rules).repeated = {max_items: 1000}];
```
Suppose I want to match:
1. Response status should be either 301 OR 302 OR 307

2. Response status is 200 AND content-type is text/html

For case 1, I would want that if ANY header KV pair matches, match the condition and rewrite the requested header.
For case 2, I would want that if ALL header KV pairs match, only then match the condition, and rewrite the requested header.

For this reason, I have introduced a list of acceptable response codes ANY of which could match.
I have also added a list of headers ALL of which must match.
Header rewrite will happen when both the subconditions match.

Reply:
My question was more around whether you could use `AccessLogFilter`, which is more general purpose than what you currently have.